uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/PublicHoliday.tpl

```diff
@@ -24,37 +24,36 @@
         </div>
       </div>
     </div>
-    {literal}
-        <script type="text/javascript">
-            CRM.$(function($) {
-                function initDeleteButton() {
-                    $('.crm-button-type-delete').on('click', function(e) {
-                        e.preventDefault();
-                        CRM.confirm({
-                            title: ts('Delete Public Holiday'),
-                            message: ts('Are you sure you want to delete this Public Holiday?'),
-                            options: {
-                                yes: ts('Yes'),
-                                no: ts('No')
-                            }
-                        })
-                        .on('crmConfirm:yes', deleteCallback);
-                    });
-                }
-
-                function deleteCallback() {
-                    {/literal}
-                    window.location = "{$deleteUrl}";
-                    {literal}
-                }
+    <script type="text/javascript">
+      {literal}
+        CRM.$(function($) {
+          var REDIRECT_AFTER_DELETE_URL = {/literal}'{$deleteUrl}'{literal};
 
-                $(document).ready(function() {
-                    initDeleteButton();
-                });
+          $(document).ready(function() {
+            initDeleteButton();
+          });
 
+          function initDeleteButton() {
+            $('.crm-button-type-delete').on('click', function(e) {
+              e.preventDefault();
+              CRM.confirm({
+                title: ts('Delete Public Holiday'),
+                message: ts('Are you sure you want to delete this Public Holiday?'),
+                options: {
+                  yes: ts('Yes'),
+                  no: ts('No')
+                }
+              })
+              .on('crmConfirm:yes', REDIRECT_AFTER_DELETE_URL);
```

Comment:
No need for `window.location = deleteUrl`? , So the URL can be passed directly?

Reply:
Not quite sure I understand. We have a constant REDIRECT_AFTER_DELETE_URL because it is better for readability so it does not mix smarty and JS too much.

Reply:
The initial implementation is this: https://github.com/compucorp/civihr/blob/e8fd6ef4adbec0b4a3f190b50da54c52677601e2/uk.co.compucorp.civicrm.hrleaveandabsences/templates/CRM/HRLeaveAndAbsences/Form/PublicHoliday.tpl#L41, It is passed a callback that that redirects to the callback URL. Now the callback URL is passed directly.

Reply:
Oh snap! Fixing.

Reply:
Fixed